app/storage/metadata_parser.py

```diff
@@ -1,105 +1,123 @@
 import uuid
 from datetime import datetime
-
 from structlog import get_logger
 
 from sdc.crypto.exceptions import InvalidTokenException
 
 logger = get_logger()
 
 
-def iso_8601_data_parser(iso_8601_string):
+def iso_8601_date_parser(iso_8601_string):
     return datetime.strptime(iso_8601_string, '%Y-%m-%d')
 
 
 def uuid_4_parser(plain_string):
     return str(uuid.UUID(plain_string))
 
 
+def clean_leading_trailing_spaces(metadata):
```

Comment:
I don't think there's any need to build a new object in this case:

```
for key, value in metadata.items():
    if isinstance(value, str):
        metadata[key] = value.strip()

return metadata
```

Reply:
Existing code, also thought it was bad practice to modify passed in arguments?.

Reply:
It's a judgement call.

Probably a bad idea in most cases but I think it's pretty explanatory here.